doc/guides/writing_tests.md

```diff
@@ -0,0 +1,166 @@
+# How to write a Node.js test
+
+## What is a test?
+
+A test must be a node script that exercises a specific functionality provided by node and
+checks that it behaves as expected. It should return 0 on success, otherwise it will
+fail. A test will fail if:
+
+- It exits by calling `process.exit(code)` where `code != 0`
+- It exits due to an uncaught exception.
+- It never exits. In this case, the test runner will terminate the test because it sets a maximum time limit.
+
+They can be added for multiple reasons:
```

Comment:
nit: I’d prefer `s/They/Tests/` because there’s no plural `tests` before this in the text.